### ISSUE TITLE
REST API - No such entity when adding items to cart - /carts/mine/items 

### ISSUE THREAD

**Issue by thiagosantos** [magento/magento2#14925]
Lost my day on this issue, basically when trying to manage a cart after added a new item, and the only one. Got repeatedly the error exposed below. Going futher I got that this error was happening because "region id" wasn't set up.

NoSuchEntityException is so generically that a error occurring inside the method, of interface, of another interface, of class, from an object, with a unset ID was causing a issue that was almost impossible to find.

There are three good options to go through this: alter **NoSuchEntityException** to bring the class and the php line where the **throws** were called (or bring real detailed information), obligate the **region id** field when adding or updating an address on API (cause in normal conditions like the shipping/billing form and in the admin this field is required) or check for the **region id** before its use.

In Love with Magento 2! :)
Best regards,
### Preconditions

1. Magento CE 2.2.2 
2. PHP 7.2
3. MariaDB 10.0.31

### Steps to reproduce

1. Using the REST API - create a customer with address but without the region ID
2. Using the REST API - create a cart - [POST] /rest/default/V1/carts/mine
3. Using the REST API - add a item do cart - [POST] /rest/default/V1/carts/mine/items
4. Then try to add a new product (go back to step 3) or try to get cart - [GET] /rest/default/V1/carts/mine

### Expected result

1. When adding a second item return the payload
2. When getting the cart return actual cart state

### Actual result

1. Always returning
```
{
"message": "No such entity with %fieldName = %fieldValue",
"parameters":{
"fieldName": "id",
"fieldValue": "4"
}
}
```

**Comment by ghost** [comment]
Hi @thiagosantos the fix for this issue has already available in 2.2.5 release

**Comment by thiagosantos** [comment]
That's sounds great!
But what solution you guys implemented?
